backend/main.py

```diff
@@ -1,11 +1,14 @@
 import os
 from fastapi import FastAPI, Query
 from flagsmith import Flagsmith
+from flagsmith.offline_handlers import LocalFileHandler
 
 
+local_file_handler = LocalFileHandler(environment_document_path="/app/flags.json")
 flagsmith = Flagsmith(
     environment_key=os.getenv("FLAGSMITH_API_KEY"),
-    enable_local_evaluation=True)
+    enable_local_evaluation=True,
+    offline_handler=local_file_handler)
```

Comment:
If there are no access to Flagsmith, I get error even though offline_handler is set:
```
feature_backend  | Traceback (most recent call last):
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 196, in _new_conn
feature_backend  |     sock = connection.create_connection(
feature_backend  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 85, in create_connection
feature_backend  |     raise err
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 73, in create_connection
feature_backend  |     sock.connect(sa)
feature_backend  | ConnectionRefusedError: [Errno 111] Connection refused
feature_backend  | 
feature_backend  | The above exception was the direct cause of the following exception:
feature_backend  | 
feature_backend  | Traceback (most recent call last):
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 789, in urlopen
feature_backend  |     response = self._make_request(
feature_backend  |                ^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 490, in _make_request
feature_backend  |     raise new_e
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 466, in _make_request
feature_backend  |     self._validate_conn(conn)
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 1095, in _validate_conn
feature_backend  |     conn.connect()
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 615, in connect
feature_backend  |     self.sock = sock = self._new_conn()
feature_backend  |                        ^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 211, in _new_conn
feature_backend  |     raise NewConnectionError(
feature_backend  | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0xffffb88854d0>: Failed to establish a new connection: [Errno 111] Connection refused
feature_backend  | 
feature_backend  | The above exception was the direct cause of the following exception:
feature_backend  | 
feature_backend  | Traceback (most recent call last):
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 667, in send
feature_backend  |     resp = conn.urlopen(
feature_backend  |            ^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 873, in urlopen
feature_backend  |     return self.urlopen(
feature_backend  |            ^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 873, in urlopen
feature_backend  |     return self.urlopen(
feature_backend  |            ^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 873, in urlopen
feature_backend  |     return self.urlopen(
feature_backend  |            ^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 843, in urlopen
feature_backend  |     retries = retries.increment(
feature_backend  |               ^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/urllib3/util/retry.py", line 519, in increment
feature_backend  |     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
feature_backend  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  | urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='edge.api.flagsmith.com', port=443): Max retries exceeded with url: /api/v1/environment-document/ (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0xffffb88854d0>: Failed to establish a new connection: [Errno 111] Connection refused'))
feature_backend  | 
feature_backend  | During handling of the above exception, another exception occurred:
feature_backend  | 
feature_backend  | Traceback (most recent call last):
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/flagsmith/flagsmith.py", line 373, in _get_json_response
feature_backend  |     response = request_method(
feature_backend  |                ^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 602, in get
feature_backend  |     return self.request("GET", url, **kwargs)
feature_backend  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 589, in request
feature_backend  |     resp = self.send(prep, **send_kwargs)
feature_backend  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 703, in send
feature_backend  |     r = adapter.send(request, **kwargs)
feature_backend  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 700, in send
feature_backend  |     raise ConnectionError(e, request=request)
feature_backend  | requests.exceptions.ConnectionError: HTTPSConnectionPool(host='edge.api.flagsmith.com', port=443): Max retries exceeded with url: /api/v1/environment-document/ (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0xffffb88854d0>: Failed to establish a new connection: [Errno 111] Connection refused'))
feature_backend  | 
feature_backend  | The above exception was the direct cause of the following exception:
feature_backend  | 
feature_backend  | Traceback (most recent call last):
feature_backend  |   File "/usr/local/lib/python3.11/multiprocessing/process.py", line 314, in _bootstrap
feature_backend  |     self.run()
feature_backend  |   File "/usr/local/lib/python3.11/multiprocessing/process.py", line 108, in run
feature_backend  |     self._target(*self._args, **self._kwargs)
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/_subprocess.py", line 80, in subprocess_started
feature_backend  |     target(sockets=sockets)
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/server.py", line 65, in run
feature_backend  |     return asyncio.run(self.serve(sockets=sockets))
feature_backend  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
feature_backend  |     return runner.run(main)
feature_backend  |            ^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
feature_backend  |     return self._loop.run_until_complete(task)
feature_backend  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
feature_backend  |     return future.result()
feature_backend  |            ^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/server.py", line 69, in serve
feature_backend  |     await self._serve(sockets)
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/server.py", line 76, in _serve
feature_backend  |     config.load()
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/config.py", line 434, in load
feature_backend  |     self.loaded_app = import_from_string(self.app)
feature_backend  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/importer.py", line 19, in import_from_string
feature_backend  |     module = importlib.import_module(module_str)
feature_backend  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
feature_backend  |     return _bootstrap._gcd_import(name[level:], package, level)
feature_backend  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
feature_backend  |   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
feature_backend  |   File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
feature_backend  |   File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
feature_backend  |   File "<frozen importlib._bootstrap_external>", line 940, in exec_module
feature_backend  |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
feature_backend  |   File "/app/backend/main.py", line 8, in <module>
feature_backend  |     flagsmith = Flagsmith(
feature_backend  |                 ^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/flagsmith/flagsmith.py", line 163, in __init__
feature_backend  |     self._initialise_local_evaluation()
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/flagsmith/flagsmith.py", line 189, in _initialise_local_evaluation
feature_backend  |     self.update_environment()
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/flagsmith/flagsmith.py", line 284, in update_environment
feature_backend  |     self._environment = self._get_environment_from_api()
feature_backend  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/flagsmith/flagsmith.py", line 296, in _get_environment_from_api
feature_backend  |     environment_data = self._get_json_response(self.environment_url, method="GET")
feature_backend  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
feature_backend  |   File "/usr/local/lib/python3.11/site-packages/flagsmith/flagsmith.py", line 383, in _get_json_response
feature_backend  |     raise FlagsmithAPIError(
feature_backend  | flagsmith.exceptions.FlagsmithAPIError: Unable to get valid response from Flagsmith API.
```

Reply:
Will be fixed here https://github.com/Flagsmith/flagsmith-python-client/pull/100